sql/core/src/test/scala/org/apache/spark/sql/util/DataFrameCallbackSuite.scala

```diff
@@ -89,38 +91,46 @@ class DataFrameCallbackSuite extends QueryTest with SharedSparkSession {
   }
 
   test("get numRows metrics by callback") {
-    val metrics = ArrayBuffer.empty[Long]
-    val listener = new QueryExecutionListener {
-      // Only test successful case here, so no need to implement `onFailure`
-      override def onFailure(funcName: String, qe: QueryExecution, error: Throwable): Unit = {}
-
-      override def onSuccess(funcName: String, qe: QueryExecution, duration: Long): Unit = {
-        val metric = qe.executedPlan match {
-          case w: WholeStageCodegenExec => w.child.longMetric("numOutputRows")
-          case other => other.longMetric("numOutputRows")
+    withSQLConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true") {
+      // with AQE on, the WholeStageCodegen rule is applied when running QueryStageExec.
+      val metrics = ArrayBuffer.empty[Long]
+      val listener = new QueryExecutionListener {
+        // Only test successful case here, so no need to implement `onFailure`
+        override def onFailure(funcName: String, qe: QueryExecution, error: Throwable): Unit = {}
+
+        override def onSuccess(funcName: String, qe: QueryExecution, duration: Long): Unit = {
+          val plan = qe.executedPlan match {
+            case a: AdaptiveSparkPlanExec => a.executedPlan
```

Comment:
This appears many times, we can introduce a `stripAQEPlan` method in `AdaptiveSparkPlanHelper` to do it.